pkg/trace/otel.go

```diff
@@ -224,6 +225,17 @@ func (t *otelTracer) addInfo(ctx context.Context, args ...log.Marshaler) {
 		for _, arg := range args {
 			kvs := marshalToKeyValue(arg)
 			span.SetAttributes(kvs...)
+
+			// setting the code will cause error spans to be called out specifically
+			// in the trace, and will make the "errors" default view in honeycomb useful
+			switch v := arg.(type) {
+			case *events.ErrorInfo:
+				span.SetStatus(codes.Error, v.Error)
+			case error:
+				span.SetStatus(codes.Error, v.Error())
```

Comment:
`error` doesn't meet the requirements of `log.Marshaler`, right?  Therefore `arg.(type)` will never be `error` and we should not hit this branch?

Reply:
maybe a comment is in order-- nothing stops clients from defining an `error` that implements `log.Marshaler`, so this is trying to catch that case, since i think in that circumstance it's reasonable for the client to expect the error to be respected.